R/validate-landings.R

```diff
@@ -116,7 +132,21 @@ validate_landings <- function(log_threshold = logger::DEBUG){
       landing_date = .data$date,
       trip_duration = .data$trip_duration,
       landed_catch = .data$species_group,
-      landed_value = .data$total_catch_value)
+      landed_value = .data$total_catch_value) %>%
+    dplyr::select(-.data$landed_catch, tidyselect::everything()) %>%
+    tidyr::unnest(.data$landed_catch, keep_empty = TRUE) %>%
+    tidyr::unnest(.data$length_frequency, keep_empty = TRUE) %>%
+    dplyr::left_join(weight_table) %>%
+    dplyr::mutate(tot_weight = (.data$a * .data$length^.data$b) * individuals) %>%
+    tidyr::nest(length_frequency = c(.data$length:.data$tot_weight)) %>%
+    dplyr::mutate(tot_weight = purrr::map_dbl(.data$length_frequency, ~ {
+      sum(.x$tot_weight, na.rm = TRUE)
+    })) %>%
+    tidyr::nest(landed_catch = c(.data$catch_taxon:.data$tot_weight)) %>%
+    dplyr::mutate(tot_weight = purrr::map_dbl(.data$landed_catch, ~ {
+      sum(.x$tot_weight, na.rm = TRUE)
```

Comment:
@efcaguab I really don't like that, I'm sure you have an easy purrr-way to do it smarter